plugin.rb

```diff
@@ -108,30 +108,32 @@ def fetch_user_details(token, id)
     log("user_json_url: #{user_json_method} #{user_json_url}")
 
     bearer_token = "Bearer #{token}"
-    user_json_response =
-      if user_json_method.downcase.to_sym == :post
-        Net::HTTP
-          .post_form(URI(user_json_url), 'Authorization' => bearer_token)
-          .body
-      else
-        Excon.get(user_json_url, headers: { 'Authorization' => bearer_token, 'Accept' => 'application/json' }, expects: [200]).body
+    connection = Excon.new(
+      user_json_url,
+      :headers => { 'Authorization' => bearer_token, 'Accept' => 'application/json' }
+    )
+    user_json_response = connection.request(method: user_json_method)
+
+    log("user_json_response: #{user_json_response.inspect}")
+
+    if user_json_response.status == 200
+      user_json = JSON.parse(user_json_response.body)
+
+      log("user_json: #{user_json}")
+
+      result = {}
+      if user_json.present?
+        json_walk(result, user_json, :user_id)
+        json_walk(result, user_json, :username)
+        json_walk(result, user_json, :name)
+        json_walk(result, user_json, :email)
+        json_walk(result, user_json, :email_verified)
+        json_walk(result, user_json, :avatar)
       end
-
-    user_json = JSON.parse(user_json_response)
-
-    log("user_json: #{user_json}")
-
-    result = {}
-    if user_json.present?
-      json_walk(result, user_json, :user_id)
-      json_walk(result, user_json, :username)
-      json_walk(result, user_json, :name)
-      json_walk(result, user_json, :email)
-      json_walk(result, user_json, :email_verified)
-      json_walk(result, user_json, :avatar)
+      result
+    else
+      false
```

Comment:
If the success path returns a hash object, then the failure path should return `nil`. Otherwise we're mixing booleans/objects for the response type.

Reply:
Indeed! Changed :+1: